karton/classifier/classifier.py

```diff
@@ -105,7 +114,13 @@ def _get_extension(self, name: str) -> str:
     def _classify(self, task: Task) -> Optional[Dict[str, str]]:
         sample = task.get_resource("sample")
         content = cast(bytes, sample.content)
-        magic = task.get_payload("magic") or pymagic.from_buffer(content)
+
+        magic = task.get_payload("magic") or ""
+        try:
+            magic = pymagic.from_buffer(content)
+        except Exception as ex:
+            self.log.exception(ex)
```

Comment:
`log.exception` logs exception from the current contxt anyway.

```suggestion
            self.log.exception("Failed to get magic")
```

Reply:
Good point. Since now this case is not fatal anymore, I'd suggest to change it to this:
```
self.log.warning(f'unable to get magic: {ex}')
```

It is useful to know which exception it was, since that will provide some details for finding the root cause.

Reply:
@conitrade-as: I see that python-magic maps the libmagic exceptions into `MagicException` class (https://github.com/ahupp/python-magic/blob/e0ccc6d0cbf7b17e8c7288297c717f3b3d499679/magic/__init__.py#L32). Maybe we should treat other exceptions as fatal and catch only the `MagicException`?

The other thing to consider: `WARNING` level sounds good enough, but we can still provide full info about exception via `exc_info` parameter (https://docs.python.org/3/library/logging.html#logging.Logger.debug). Libmagic faults are not so often, so it shouldn't be too noisy.

What do you think about these changes?

Reply:
There are other things which can bubble up. E.g. run the tests on a Mac and you will see an instance of `UnicodeDecodeError`. In this case the log entry will look like this:
```
unable to get magic: 'utf-8' codec can't decode byte 0xca in position 102: invalid continuation byte
```

Reply:
Hmm, that sounds like a bug :thinking: python-magic uses `maybe_decode` which converts data back to `str` with `errors=backslashreplace`. Are you sure it's platform-specific, not Python version specific? (e.g. < 3.6)

Nevertheless I agree that we can still try to handle it gracefully and fallback to the empty magic.
```python
        try:
            magic = pymagic.from_buffer(content)
        except Exception:
            self.log.warning("Unable to get magic", exc_info=True)
 ```

Reply:
This is what I have on my Mac:
```
% python
Python 3.8.2 (default, Dec 21 2020, 15:06:04) 
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
```

Reply:
Btw. the document in question is the following: `document:win32:doc-12a_a.doc`.